src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,223 @@
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mappings_df['source_prefix'] = mappings_df['source_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['source_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['source_prefix']
+    # - filter obsoletes (from mapping set --> source & mondo)
+    #  - filter obsoleted in Mondo
+    mappings_df = mappings_df[~mappings_df['mondo_label'].str.startswith('obsolete')]
+    #  - filter obsoleted in source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_obsoletes: Set[CURIE] = set([x for x in source_db.obsoletes()])
+    mappings_df = mappings_df[~mappings_df['source_id'].isin(source_obsoletes)]
+    # - update subject_labels
+    source_labels: List[Tuple[CURIE, str]] = [x for x in source_db.labels(mappings_df['source_id'].tolist())]
+    source_labels: Dict[CURIE, str] = {curie: label for curie, label in source_labels}
+    mappings_df['source_label'] = mappings_df['source_id'].map(source_labels)
+
+    # Query synonyms: source
+    source_df: pd.DataFrame = _query_synonyms(mappings_df['source_id'].tolist(), source_db)
+    # - add mapped mondo ids
+    source_mondo_lookup = mappings_df.set_index('source_id')['mondo_id'].to_dict()
+    source_df['mondo_id'] = source_df['curie'].map(source_mondo_lookup)
+    source_df = source_df.rename(columns={'curie': 'source_id'})[['mondo_id', 'source_id', 'synonym_scope', 'synonym']]
+    # - add labels
+    source_df = source_df.merge(mappings_df, on=['mondo_id', 'source_id'], how='left')
+    # TODO: #0: perhaps filter / check if some of the mapped source IDs are deleted, and filter out of mapping_df etc
+    #  finally: this could be an issue for other pipelines, like subclass sync
+
+    # Query synonyms: Mondo
+    mondo_df = pd.read_csv(mondo_synonyms_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    # - filter by source
+    mondo_df['source_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['source_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['source_prefix']
+    # filter out any non-source-ID, source-specific synonym xrefs
+    # Todo: this should statically just filter OMIM:genemap2 and nothing else
+    mondo_df['unprefixed_source_id'] = mondo_df['source_id'].apply(lambda x: x.split(':')[1])
+    mondo_df = mondo_df[mondo_df['unprefixed_source_id'].str.isnumeric()]
+    del mondo_df['unprefixed_source_id']
+    # - add labels
+    mondo_df = mondo_df.merge(mappings_df, on=['mondo_id', 'source_id'], how='left')
+
+    # TODO: (#0, #1, etc) why do NA / 'no labels' exist?
+    # print(any(list(source_df['mondo_label'].isna())))  # f
+    # print(any(list(source_df['source_label'].isna())))  # f
+    # print(any(list(mappings_df['mondo_label'].isna())))  # f
+
+    # TODO: #0 deleted sources (source_label na)
```

Comment:
## Deleted terms
This is similar to obsoleted terms, but not quite the same. I'm encountering a data quality issue where we have mappings in `mondo.sssom.tsv` that are mapped to terms which no longer exist in the source ontology.

- [ ] My aim is to filter these out

Reply:
Done via:
```py
    # - filter mapped but deleted IDs
    #   If there is no label, we take that as a proxy / sign that it has been deleted. Alternative ways: entities(),
    #   all_entity_curies(), node() (empty props), node_exists() (no: falsely says 'True', prolly cuz bad xrefs exist)
    source_labels: List[Tuple[CURIE, str]] = [x for x in source_db.labels(mappings_df['source_id'].tolist())]
    source_deleted_ids: Set[CURIE] = set([x[0] for x in source_labels if not x[1]])
    mappings_df = mappings_df[~mappings_df['source_id'].isin(source_deleted_ids)]
```

Reply:
@twhetzel You can mark this resolved when you see it. Just letting you know it's done.

Reply:
I need more explanation of the data quality issue here and an example row found in the `mondo.sssom.tsv` file.

Reply:
@twhetzel We discussed last week; the examples I have are all OMIMPS. I don't think I looked at other sources, so perhaps the problem is broader than that. Here's the issue I opened up about it.
- https://github.com/monarch-initiative/mondo/issues/7955